Source/WebCore/rendering/style/ShadowData.cpp

```diff
@@ -38,6 +38,15 @@ ShadowData::ShadowData(const ShadowData& o)
 {
 }
 
+void ShadowData::deleteNextLinkedListWithoutRecursion()
+{
+    // Avoid recursion errors when the linked list is too long.
+    Vector<std::unique_ptr<ShadowData>> list;
+    list.append(WTFMove(m_next));
+    while (auto& next = list.last()->m_next)
```

Comment:
I think you can do this without accumulating a Vector. Something like:

for (auto next = std::exchange(m_next, nullptr); next = std::exchange(next->m_next, nullptr)) { }

Reply:
I applied the suggestion. Definitely more efficient this way.